posthog/hogql/parser.py

```diff
@@ -1120,8 +1120,8 @@ def visitHogqlxTagAttribute(self, ctx: HogQLParser.HogqlxTagAttributeContext):
             return ast.HogQLXAttribute(name=name, value=ast.Constant(value=True))
 
     def visitPlaceholder(self, ctx: HogQLParser.PlaceholderContext):
-        name = self.visit(ctx.identifier())
-        return ast.Placeholder(field=name)
+        nested = self.visit(ctx.nestedIdentifier()) if ctx.nestedIdentifier() else []
+        return ast.Placeholder(field=".".join(identifier for identifier in nested))
```

Comment:
If we're going the nested identifiers route, I propose we make `Placeholder` work just like `Field`, ie. turn `field: str` into `chain: list[str]`.
That requires updating a few other places in a minor way, but importantly prevents all the `.join()`ing and `.startswith()`ing, which are a bit smelly in this context. E.g. `placeholder.startswith("foo.")` is less trustworthy than `placeholder[:1] == "foo"`, as the former is less rigid (someone can even forget the ".")